### PR TITLE
Aws task7 lambda basic authorization

### PR DESCRIPTION
**Pull request by colpachoque**
**TASK_7.1**
- [x] Create new service authorization-service
- [x] Create basicAuthorizer lambda
 
**TASK_7.2**
- [x] Add Lambda authorization (basicAuthorizer lambda) to the /import path of the import-service API Gateway.

**TASK_7.3**
- [x] Request from the client application to the /import path of the import-service should have Basic Authorization header
- [x] Client should get authorization_token value from browser localStorage (necessary changes are done in fe repo)

**Additional task**
- [x] alerts added on fe

(commit with changes - https://github.com/rolling-scopes-school/nodejs-aws-fe/commit/cacf75fa62bbae71689e709ceeb042cf9a7235e8)

My username - colpachoque
Correct token (sorry, I named it 'token' not 'authorization_token') - `localStorage.setItem('token', 'Y29scGFjaG9xdWU6VEVTVF9QQVNTV09SRA==');`

Link to the [app](https://dqx938xovkpeh.cloudfront.net/admin/products)
Link for testing import lambda via postman - https://m2aacpogv2.execute-api.eu-west-1.amazonaws.com/dev/import

Screenshot with successful authorization
Снимок экрана 2020-11-29 в 14 46 55

Screenshot with 403
Снимок экрана 2020-11-29 в 14 47 13

Screenshot with 401
Снимок экрана 2020-11-29 в 14 47 24

Screenshot with alert on FE app page
image

### CODE REVIEW

File: another-import-service/serverless.yml [hunk 1]
```diff
@@ -43,6 +46,10 @@ functions:
               querystrings:
                 name: true
           cors: true
+          authorizer:
+            arn: arn:aws:lambda:eu-west-1:517465834650:function:authorization-service-dev-basic-authorizer
```
Comment (IvanPresmytsky) [inline another-import-service/serverless.yml:70]: What about using export/import for this arn instead of hardcoding it? :)

**Review by IvanPresmytsky — APPROVED** [review]

**Review by IvanPresmytsky — APPROVED** [review]
good job! But there is one small issue.
I succeeded to make a request by postman, but the request was failed on FE because of CORS.

File: another-import-service/serverless.yml [hunk 2]
```diff
@@ -30,6 +30,9 @@ provider:
         - 'sqs:*'
       Resource:
         - 'arn:aws:sqs:eu-west-1:517465834650:catalogItemsQueue'
+  environment:
+    SQS_URL:
+      Fn::ImportValue: SQSUrl
```
Comment (IvanPresmytsky) [inline another-import-service/serverless.yml:56]: The issue with CORS could be solved this way:

```
resources:
    Resources: 
      GatewayResponseAccessDenied: 
        Type: 'AWS::ApiGateway::GatewayResponse',
        Properties: 
          RestApiId: 
            Ref: 'ApiGatewayRestApi'
          ResponseType: 'ACCESS_DENIED',
          ResponseParameters:
            'gatewayresponse.header.Access-Control-Allow-Origin': "'*'",
            'gatewayresponse.header.Access-Control-Allow-Headers': "'*'",
      GatewayResponseUnauthorized:
        Type: 'AWS::ApiGateway::GatewayResponse',
        Properties:
          RestApiId:
            Ref: 'ApiGatewayRestApi'
          ResponseType: 'UNAUTHORIZED',
          ResponseParameters:
            'gatewayresponse.header.Access-Control-Allow-Origin': "'*'",
            'gatewayresponse.header.Access-Control-Allow-Headers': "'*'",
```

Comment (colpachoque) [reply another-import-service/serverless.yml:56]: Hmm, I checked it and it worked on the FE as well (I mean new imported products were successfully created)... But thank you for this catch, will recheck it!

Comment (colpachoque) [reply another-import-service/serverless.yml:56]: Fixed, thank you!

**Comment by Emerkof** [comment]
@colpachoque 

Getting a `403 Forbidden` error, when the `token` is not provided. Should be `401 Unauthorized` instead:

image

Quite a minor issue though, so giving a max score anyway

**Pull request merged**